examples/vision/semisupervised_simclr.py

```diff
@@ -0,0 +1,743 @@
+            ),
+        )
+
+    def random_contrast(self, images):
+        # random interpolation/extrapolation between the image and its mean intensity value
+        mean = tf.reduce_mean(
+            tf.image.rgb_to_grayscale(images), axis=(1, 2), keepdims=True
+        )
+        return self.blend(
+            images,
+            mean,
+            tf.random.uniform(
+                (tf.shape(images)[0], 1, 1, 1), 1 - self.contrast, 1 + self.contrast
+            ),
+        )
+
+    def random_saturation(self, images):
+        # random interpolation/extrapolation between the image and its grayscale counterpart
+        return self.blend(
+            images,
+            tf.image.rgb_to_grayscale(images),
+            tf.random.uniform(
+                (tf.shape(images)[0], 1, 1, 1), 1 - self.saturation, 1 + self.saturation
+            ),
+        )
+
+    def random_hue(self, images):
+        # random shift in hue in hsv colorspace
+        images = tf.image.rgb_to_hsv(images)
+        images += tf.random.uniform(
+            (tf.shape(images)[0], 1, 1, 3), (-self.hue, 0, 0), (self.hue, 0, 0)
+        )
+        # tf.math.floormod(images, 1.0) should be used here, however in introduces artifacts
+        images = tf.where(images < 0.0, images + 1.0, images)
+        images = tf.where(images > 1.0, images - 1.0, images)
+        images = tf.image.hsv_to_rgb(images)
+        return images
+
+    def call(self, images, training=True):
+        if training:
+            # applies color augmentations in random order
+            for color_augmentation in random.sample(self.color_augmentations, 4):
+                images = color_augmentation(images)
+        return images
+
+
+"""
+## Image augmentation modules
+
+We use stronger augmentations for contrastive learning, along with weaker ones for
+supervised classification to avoid overfitting on the few labeled examples.
+"""
+
+# stronger augmentations are used for the contrastive task
+def get_contrastive_augmenter():
+    return keras.Sequential(
+        [
+            layers.Input(shape=(96, 96, 3)),
+            preprocessing.Rescaling(1 / 255),
+            preprocessing.RandomFlip("horizontal"),
+            RandomResizedCrop(scale=(0.2, 1.0), ratio=(3 / 4, 4 / 3)),
+            RandomColorJitter(brightness=0.5, contrast=0.5, saturation=0.5, hue=0.2),
+        ],
+        name="contrastive_augmenter",
+    )
+
+
+# weaker augmentations are used for the classification task
+def get_classification_augmenter():
+    return keras.Sequential(
+        [
+            layers.Input(shape=(96, 96, 3)),
+            preprocessing.Rescaling(1 / 255),
+            preprocessing.RandomFlip("horizontal"),
+            RandomResizedCrop(scale=(0.5, 1.0), ratio=(3 / 4, 4 / 3)),
+            RandomColorJitter(brightness=0.2, contrast=0.2, saturation=0.2, hue=0.1),
+        ],
+        name="classification_augmenter",
+    )
+
+
```

Comment:
I would recommend refactoring get_classification_augmenter() and get_contrastive_augmenter() as the only difference is the ResizeCrop and ColorJitter color. That's a lot duplicated code. 

I would also try to not have the parameters hardcoded so maybe you can have a dictionary object 
`classification_augmentation = {
"shape": (96, 96, 3),
....
"resizedcrop_scale": (0.5, 1.0)
...
}`

Reply:
Good point, I refactored them to a single function, with customizable augmentation parameters. I would however prefer to keep the input sizes hardcoded, for simplicity.

Reply:
hardcoded inputs makes the code harder to reuse for different sizes. You would have to go and manually replace variables through the code. If you like to make it easier to read you could use:
`WIDTH = 256` or something like this at the top of the example so its clear those are constants and you see where they are used.